src/lazy/binary/raw/v1_1/value.rs

```diff
@@ -119,7 +119,13 @@ impl<'top> HasRange for &'top LazyRawBinaryValue_1_1<'top> {
 
 impl<'top> LazyRawValue<'top, BinaryEncoding_1_1> for &'top LazyRawBinaryValue_1_1<'top> {
     fn ion_type(&self) -> IonType {
-        self.encoded_value.ion_type()
+        // Handle retrieving the type for a typed null.
+        if self.encoded_value.header.type_code() == OpcodeType::TypedNull {
+            let body = self.value_body();
+            ION_1_1_TYPED_NULL_TYPES[body[0] as usize]
+        } else {
+            self.encoded_value.ion_type()
+        }
```

Comment:
Could you put this functionality in a `self.read_null()` method instead of having it happen in `ion_type()`?

There are three other raw value types besides `LazyRawBinaryValue_1_1`:
* `LazyRawBinaryValue_1_0`
* `LazyRawTextValue_1_1`
* `LazyRawTextValue_1_0`

Each of them has an `ion_type()` method and I'm pretty sure at least some of them return `IonType::Null` if the value is a null value, even if it's a typed null. The behavior you've implemented is probably more correct, but if we make this change here we'd need to make it in the other three places too.

Reply:
Added #828 to track this.

Reply:
ack, ty, I will follow up with that today.